src/compiler/checker.ts

```diff
@@ -365,6 +366,13 @@ namespace ts {
             getDeclaredTypeOfSymbol,
             getPropertiesOfType,
             getPropertyOfType: (type, name) => getPropertyOfType(type, escapeLeadingUnderscores(name)),
+            getPrivateIdentifierPropertyOfType: (leftType: Type, right: PrivateIdentifier) => {

```

Comment:
The public API you're recommending sounds much, much easier to use!

What is `location` in `getPrivateIdentifierPropertyOfType: (leftType: Type, name: string, location: Node)`?

Reply:
`location` is used to lookup the private identifier in the lexical scope (since multiple private identifiers can have the same description and each declaration is unique)

Reply:
The `location` is the node from which to start traversing parent nodes to find the containing class declaration that contains the `name` of a private identifier (basically the logic you're using in `lookupSymbolForPrivateIdentifierDeclaration`).

You could basically change `lookupSymbolForPrivateIdentifierDeclaration` to accept a `name` and a `location`:

```ts
function lookupSymbolForPrivateIdentifierDeclaration(propName: __String, location: Node): Symbol | undefined {
    for (let containingClass = getContainingClass(location); !!containingClass; containingClass = getContainingClass(containingClass)) {
        const { symbol } = containingClass;
        const name = getSymbolNameForPrivateIdentifier(symbol, propName);
        const prop = (symbol.members && symbol.members.get(name)) || (symbol.exports && symbol.exports.get(name));
        if (prop) {
            return prop;
        }
    }
}
```